### PR TITLE
Webinterface for user written queries

### PR DESCRIPTION
**Pull request by Max-Leopold**
Project Board Card: https://github.com/Max-Leopold/mini-analytics/projects/1#card-25214990

### CODE REVIEW

File: query-input/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>de.brandwatch.minianalytics</groupId>
+    <artifactId>mini-analytics</artifactId>
+    <version>0.1</version>
+    <!-- lookup parent from repository -->
+  </parent>
+
+  <artifactId>query-input</artifactId>
```
Comment (YNedderhoff) [inline query-input/pom.xml]: I'd still suggest to rename this, as it's gonna be much more

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@org.springframework.stereotype.Controller
+public class Controller {
+
+    Logger logger = LoggerFactory.getLogger(Controller.class);
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: could be a constant?

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package de.brandwatch.minianalytics.queryinput.controller;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@org.springframework.stereotype.Controller
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: That's a bit unusual. Why not import it and just use `@Controller`?

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class Controller {
+
+    Logger logger = LoggerFactory.getLogger(Controller.class);
+
+
+    @RequestMapping(value = "/index")
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: You should be explicit about the request method you're using, either by adding `method= RequestMethod.GET`, or even better, just use `@GetMapping` instead.

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+@org.springframework.web.bind.annotation.RestController
+public class RestController {
+
+    Logger logger = LoggerFactory.getLogger(RestController.class);
+
+    @Autowired
+    QueryRepository queryRepository;
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: can be private

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+@org.springframework.web.bind.annotation.RestController
+public class RestController {
+
+    Logger logger = LoggerFactory.getLogger(RestController.class);
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: can be constant

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Autowired
+    QueryRepository queryRepository;
+
+    static final private Gson gson = new GsonBuilder().setPrettyPrinting().create();
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: the normal order would be `private static final`

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    @RequestMapping(value = "/queries", method = RequestMethod.POST)
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: same here, you can just use `@PostMapping` as of Spring Boot 5

Comment (YNedderhoff) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: and below for the others

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    public ResponseEntity query(@RequestParam String query){
+        logger.info("API Call: query");
+        logger.info("Query: " + query);
+
+        //TODO parse Query into Lucene Query
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: usually you would do that in a service layer. The usual architecture for apis like this is

Controller
Service
Repository

where the controller is only reponsible for handling the requests, possibly stuff like validation. The service is responsible for business logic, such as converting to a lucene query. And the repository is just responsible to provide methods to interact with any kind of storage.

This is also why Spring has the special annotations `@Controller`, `@Service`, `@Repository`. This automatically creates beans for those without you having to explicitly define them, and it helps to read code quicker because you always know where in the layer you are.

Of course not everything fits this pattern, which is why there is also the generic `@Component` I have mentioned before, or just defining the `@Bean` directly.

Long story short, this and the whole try catch block below propably belong into a `QueryService` class.

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        try {
+            Query saveQuery = new Query(query);
+            queryRepository.save(saveQuery);
+
+            logger.info("Retrieved query: " + query);
+            return ResponseEntity.status(200).body(gson.toJson(query));
```
Comment (YNedderhoff) [inline query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: It is good practice to return the persisted object in rest. You are just returning the object you have parsed the input string to. It might seem trivial, but can actually make a big difference.

The repository save method should actually return the object that it has successfully stored, so you can just use that.

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java (hunk 3 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Because the class is also named Controller. So the file wont compile with `@Controller` annotation

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 10 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 9 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 8 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 7 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 6 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java (hunk 5 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java (hunk 4 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java (hunk 2 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Done

File: query-input/pom.xml (hunk 1 above)
Comment (Max-Leopold) [reply query-input/pom.xml]: Done

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java (hunk 3 above)
Comment (YNedderhoff) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Right. I would generally advice against that kind of naming altogether, i.e. as a default don't name things like library classes or annotations. Not only because the imports, but also because those names are usually very generic, and too generic for your purpose. i.e. the class name `Controller` doesn't tell me anything about what it's doing. There can certainly be multiple classes that have the `@Controller` annotation, but do different things. 

This seems to serve the web interface, so maybe call it `WebController` or `InterfaceController` or something like that. The other one could be the `QueryController`

Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Ok, I renamed them.
But my problem is more general. I often don't know how to name things like the `Controller` or `RestController`. Is there any specific schema you name your classes?

Comment (YNedderhoff) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Not beyond I try to give things a sensible name the tells me what it is about. 

On the book shelf in the ping pong room there is a book called "Clean Code" that has a whole chapter called "Meaningful Names". Feel free to check it out ;) (also, there's other great books there ...)

File: API/src/main/java/de/brandwatch/minianalytics/queryinput/QueryInputApplication.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+package de.brandwatch.minianalytics.queryinput;
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/queryinput/QueryInputApplication.java]: You probably want to rename the package as well then?

```suggestion
package de.brandwatch.minianalytics.api;
```

File: query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java (hunk 3 above)
Comment (Max-Leopold) [reply query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java]: Definitly will have a look at it

File: API/src/main/java/de/brandwatch/minianalytics/queryinput/QueryInputApplication.java (hunk 11 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/queryinput/QueryInputApplication.java]: Yes, done. I was wondering why the mvn install still worked...thats why

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+@org.springframework.web.bind.annotation.RestController
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: No need for that, just impprt

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+public class QueryController {
+
+    private static final Logger logger = LoggerFactory.getLogger(QueryController.class);
+
+    @Autowired
+    private QueryService queryService;
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Now that I've said it in the other PR: I _always_ recommend constructor-based injection. That being said, not everyone does it, and this is not wrong, so I'll leave it up to you. I am not gonna mention it every time now.

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/WebController.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
+package de.brandwatch.minianalytics.api.controller;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class WebController {
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/WebController.java]: a few too many empty lines in this file

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 12 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Changed

File: API/src/main/java/de/brandwatch/minianalytics/api/service/QueryService.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    private QueryRepository queryRepository;
+
+    public ResponseEntity createQuery(String query){
+        try {
+            Query saveQuery = new Query(query);
+
+            logger.info("Retrieved query: " + query);
+            return ResponseEntity.status(200).body(gson.toJson(queryRepository.save(saveQuery)));
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/service/QueryService.java]: looking at it like this, I think it might be a little unconventional to return a ResponseEntity here instead of a query. Same of course for the other methods. Especially if you consider that you might wanna use this service for something else later. Let's assume you have a MentionController to fetch mentions for a given query, but first you want to validate if the query exists and use this Service for that.

It might be better to just throw a meaningful exception in the catch and handle the HTTP Status with some exception handling solution. There's multiple of those in Spring: https://www.baeldung.com/exception-handling-for-rest-with-spring

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+        this.queryService = queryService;
+    }
+
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    @PostMapping(value = "/queries")
+    public ResponseEntity query(@RequestParam String query){
+        try {
+            logger.info("POST API Call: /query");
+            logger.info("Query: " + query);
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I'd suggest to put these into one log. Also, never concatenate like this, instead, do 
```suggestion
            logger.info("Query: {}", query);
```

I'd suggest:

```
logger.info("POST /query: {}", query);
```

or if you want also something like this:

```
logger.info("'POST /query' api call with query string '{}'", query);
```

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+            return ResponseEntity.status(200).body(queryService.createQuery(query));
+        }catch (Exception e){
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
+        }
+
+    }
+
+    @GetMapping(value = "/queries")
+    public ResponseEntity queries(){
+        logger.info("GET API Call: /queries");
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I'd adjust this format to whatever you choose for the POST

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+        try{
+            return ResponseEntity.status(200).body(queryService.getAllQueries());
+        }catch (Exception e){
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
+        }
+    }
+
+    @GetMapping(value = "/queries/{queryID}")
+    public ResponseEntity singleQuery(@PathVariable("queryID") String queryID){
+        logger.info("GET API Call: /queries/" + queryID);
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Same as above re formatting and concatenation

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 16 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I reformatted the logger outputs.

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+        this.queryService = queryService;
+    }
+
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    @PostMapping(value = "/queries")
+    public ResponseEntity query(@RequestParam String query){
+        try {
+            logger.info("POST /queries: {}", query);
+
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: no need for this empty line

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+        }catch (Exception e){
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
+        }
+    }
+
+    @GetMapping(value = "/queries/{queryID}")
+    public ResponseEntity singleQuery(@PathVariable("queryID") String queryID){
+        try{
+            logger.info("GET /queries/" + queryID);
+
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: no need for this empty line

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
+        }
+
+    }
+
+    @GetMapping(value = "/queries")
+    public ResponseEntity queries(){
+        try{
+            logger.info("GET /queries");
+
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: no need for this empty line

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(QueryController.class);
+
+    private QueryService queryService;
+
+    @Autowired
+    public QueryController(QueryService queryService) {
+        this.queryService = queryService;
+    }
+
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: is that used?

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+
+    }
+
+    @GetMapping(value = "/queries")
+    public ResponseEntity queries(){
+        try{
+            logger.info("GET /queries");
+
+            return ResponseEntity.status(200).body(queryService.getAllQueries());
+        }catch (Exception e){
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]:
```suggestion
        } catch (Exception e) {
```

there more like this in the file. IntelliJ formatting can help you with that (opt+command+L)

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+    public ResponseEntity singleQuery(@PathVariable("queryID") String queryID){
+        try{
+            logger.info("GET /queries/" + queryID);
+
+            return ResponseEntity.status(200).body(queryService.getQueryByID(queryID));
+        }catch (Exception e){
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
+        }
+    }
+
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: unnecessary empty lines

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 23 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: This actually only happens if I use auto formatting in IntelliJ. I don't use a whitespace after '{' but IntelliJ does. I probably have to change it somewhere.

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 24 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: So, let's talk about empty lines. For me Code is much more readable if logical parts are separated. For example in this case:

`logger.info("GET /queries/" + queryID);`

and 

`return ResponseEntity.status(200).body(queryService.getQueryByID(queryID));`

I don't know, but I have the feeling, that when I remove the line there the code is much harder to read...

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 23 above)
Comment (YNedderhoff) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: This is not about a whitespace after `{` though

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 24 above)
Comment (YNedderhoff) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I can see that for logical blocks, and I do it myself. But if you argue that every single line is a logical block, I don't agree. 

In this special case I don't mind so much though, you pick.

Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I already changed it and removed the lines, so it doesn't mater now. In the future I try to have less empty lines.

File: API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java (hunk 23 above)
Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Then I don't see the issue

Comment (YNedderhoff) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: It's missing after `}` and before `{`. You see that in the suggested change above.

Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Oh... didn't know it has to be there. Doe's it always have to be there? I normally don't do it.

Comment (YNedderhoff) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: Yes, that's the convention. And IntelliJ does it by default if you do autoformatting - I wonder how it doesn't for you.

Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: It does do it for me. Just when I write code then I don't do it. IntelliJ formatting does put a whitespace there.

Comment (Max-Leopold) [reply API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java]: I reformatted it, so it should be fine now.

File: API/src/main/java/de/brandwatch/minianalytics/api/QueryInputApplication.java [hunk 25]
```diff
@@ -0,0 +1,12 @@
+package de.brandwatch.minianalytics.api;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class QueryInputApplication {
```
Comment (YNedderhoff) [inline API/src/main/java/de/brandwatch/minianalytics/api/QueryInputApplication.java]: Old name
